sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -248,6 +249,74 @@ case class OptimizeSkewedJoin(conf: SQLConf) extends Rule[SparkPlan] {
       } else {
         smj
       }
+
+    case bnl @ BroadcastNestedLoopJoinExec(leftChild, rightChild, buildSide, joinType, _, _) =>
+      def resolveBroadcastNLJoinSkew(
```

Comment:
This is just a suggestion; could we share code between smj and bnl cases? Most parts look duplicated.

Reply:
> This is just a suggestion; could we share code between smj and bnl cases? Most parts look duplicated.

After all detail ok, I will start this work.   I quite agree with this suggestion.